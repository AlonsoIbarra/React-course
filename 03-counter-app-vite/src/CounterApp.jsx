import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const subOne = () => {
        setCounter( (newVal) => parseInt(counter) - 1  );
    }
    const reset = () => {
        setCounter( (newVal) => value  );
    }
    const addOne = () => {
        setCounter( (newVal) => parseInt(counter) + 1  );
    }
    
    return (
        <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button onClick={ subOne }>
            -
        </button>
        <button onClick={ reset }>
            0
        </button>
        <button onClick={ addOne }>
            +
        </button>
        </>
    );
}

CounterApp.propTypes = {
    'value': PropTypes.number
};

CounterApp.defaultProps = {
    'value': 0 
};