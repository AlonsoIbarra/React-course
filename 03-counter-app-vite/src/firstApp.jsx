import PropTypes from 'prop-types';

export const FirstApp = ( { title, subtitle } ) => {
    console.log(title);

    return (
        <>
            <h1>{ title }</h1>
            <h2>{ subtitle }</h2>
            <p>sdf</p>
        </>
    )
}
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}
FirstApp.defaultProps = {
    title: 'Titulo',
    subtitle: 'subtitulo',
}