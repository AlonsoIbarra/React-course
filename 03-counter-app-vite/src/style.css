html, body{
    background-color: hsl(60, 2%, 87%);
    color: black;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
}
button {
    padding: 5px;
    font-size: 1.5rem;
}