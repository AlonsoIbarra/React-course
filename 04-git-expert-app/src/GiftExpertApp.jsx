import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GiftExperApp = () =>{
    const [ categories, setCategories ] =useState( [] );

    const onAddCategory = ( newCategory )=>{
        if ( categories.includes( newCategory ) ) return;
        setCategories(
            ( categories ) => [ newCategory, ...categories ]
        );
    }

    return(
        <>
            <h1>GiftExperApp</h1>
            <AddCategory onNewCategory={ onAddCategory } />
            { categories.map(
                category =>{
                    return <GifGrid category={ category } key={ category }/>
                }
            ) }
        </>
    );
}