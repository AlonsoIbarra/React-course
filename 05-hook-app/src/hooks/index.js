export * from './useCounter';
export * from './useFetch';
export * from './useForm';
export * from './intro-reducer';
export * from './todoReducer';
export * from './useTodos';