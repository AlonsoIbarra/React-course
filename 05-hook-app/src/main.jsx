// import React from 'react'
import { BrowserRouter } from 'react-router-dom';
import ReactDOM from 'react-dom/client'
import './index.css'

// import { HooksApp } from './HooksApp.jsx'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterAppWhitCustomHook } from './01-useState/CounterWhitCustomHook'
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHook } from './03-examples/MultipleCustomHooks'
// import { FocusScreen } from './04-useRef/FocusScreen'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { Memorize } from './06-memorize/memorize'
// import { MemoHook } from './06-memorize/MemoHook'
// import { CallbackHook } from './06-memorize/CallbackHook'
// import { Padre } from './07-tarea-memo/Padre'
// import './08-useReducer/intro-reducer'
// import { TodoApp } from './08-useReducer/TodoApp'
import { MainApp } from './09-useContext/mainApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
  {/* <React.StrictMode> */}
    <MainApp/>
  {/* </React.StrictMode>, */}
  </BrowserRouter>
)
