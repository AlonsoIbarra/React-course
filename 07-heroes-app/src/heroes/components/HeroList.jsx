import { useMemo } from "react";
import { getHeroesByPublisher } from "../helpers"
import { HeroCard } from "./";

export const HeroList = ({publisher})=>{
    const herosList = useMemo(()=> getHeroesByPublisher(publisher), [ publisher]);
    return (
        <>
            <div className="row rows-cols-1 rows-cols-md-1 g-3">
                <div>Hero List</div>
                <ul>
                    {herosList.map((hero) => (
                        <HeroCard key={hero.id}  {...hero}/>
                    ))}
                </ul>
            </div>
        </>
    )
}