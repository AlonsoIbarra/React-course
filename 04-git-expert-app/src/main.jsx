import React from 'react';
import ReactDOM from 'react-dom/client';
import { GiftExperApp } from './GiftExpertApp';
import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GiftExperApp />
  </React.StrictMode>,
)
