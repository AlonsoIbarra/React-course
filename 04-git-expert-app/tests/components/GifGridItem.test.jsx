import {render, screen} from '@testing-library/react';
import { GifGridItem } from '../../src/components';

describe('Snapshot test en <GifGridItem>',()=>{
    test('Debe hacer match con snapshop',()=>{
        const props = {
            'id': 'some-id',
            'url': 'hola',
            'title' : 'hola'
        };
        const { container } = render( <GifGridItem id={props.id} {...props }/>);

        expect(container).toMatchSnapshot();

    });

    test('Mostrar imagen y url correcto' , ()=>{
        const props = {
            'id': 'some-id',
            'url': 'http://custom-url/',
            'title' : 'title'
        };
        render( <GifGridItem id={props.id} {...props }/>);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(props.url);
        expect( alt ).toBe(props.title);
    });

    test('Mostrar titulo en componente', ()=>{
        const props = {
            'id': 'some-id',
            'url': 'http://custom-url/',
            'title' : 'title'
        };
        render( <GifGridItem id={props.id} {...props }/>);

        expect(screen.getByText(props.title)).toBeTruthy();

    });
})
