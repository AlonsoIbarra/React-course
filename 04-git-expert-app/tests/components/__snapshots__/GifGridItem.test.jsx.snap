// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test en <GifGridItem> Debe hacer match con snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="hola"
      src="hola"
    />
    <p>
      hola
    </p>
  </div>
</div>
`;
